duckduckgo/iOS
# encourage users to add the app to their dock

Task/Issue URL: https://app.asana.com/0/35672304057861/619654283307010

* A/B implementation:  https://app.asana.com/0/414235014887631/640129010472305
* CTA and instructions: https://app.asana.com/0/414235014887631/640129010472306
* Reminder notification: https://app.asana.com/0/414235014887631/648387649699574 
* Variant in feedback: https://app.asana.com/0/414235014887631/653505507697886

Tech Design URL: 

* A/B: https://app.asana.com/0/392891325557410/640129010472304

CC:

**Description**:

For new installs only:
* adds a call to action on the home screen to launch a video showing how to add the app to the dock
* 3 days after the install (and if the CTA was dismissed without showing the instructions) a reminder will show in the form of a notification when the user does a search (or selects from the autocomplete)

For all installs:
* Setting screen allows user to launch instructions
* Home screen shows Dax up front and centre but keyboard is down by default
* When opening a new tab from the tab switcher it shows the keyboard
* When playing the instruction video ensure background audio is not stopped

*Useful info when testing:*

Override the default variant by replacing `currentVariant` in `DefaultVariantManager` with something that returns the variant you want directly from the array in `Variant.defaultVariants`

**Steps to test this PR**:

*Existing install and all variants:*

1. note keyboard is open by default - tap home screen to lose focus from omni bar
1. go to settings and tap "Add DuckDuckGo to your dock" - instructions are launched
1. Check looks OK in landscape and portrait
1. Check other settings options still work as expected
1. Tap send feedback and ensure current variant is in the body (or "-" if existing install/no variant)

*With an existing install:*

1. launch the app - no CTA 
1. forward the device's date by 1 day and perform a search - no reminder
1. forward the device's date by 2 days and perform a search - no reminder
1. forward the device's date by 3 days and perform a search - no reminder
1. check that atb sent with requests has not been altered

*New install - m1 variant*

1. Should behave the same as an existing install.
1. check that atb sent with requests has includes m1 variant

*New install - m2 variant*

1. launch the app - after onboarding CTA should be on home screen
1. tap the CTA - the instructions launch
1. forward the device's date by 1 day and perform a search - no reminder
1. forward the device's date by 2 days and perform a search - no reminder
1. forward the device's date by 3 days and perform a search - no reminder
1. repeat above but dismiss CTA instead of tapping it
1. check that atb sent with requests has includes m2 variant

*New install - m3 variant* 

1. launch the app - after onboarding CTA should be on home screen
1. dismiss the CTA 
1. forward the device's date by 1 day and perform a search - no reminder
1. forward the device's date by 2 days and perform a search - no reminder
1. forward the device's date by 3 days and perform a search - reminder is shown
1. check looks OK in landscape and portrait
1. repeat above fresh install but tap CTA instead of tapping it - reminder should not be shown
1. check subsequent searches don't show reminder
1. check that atb sent with requests has includes m3 variant

*New install - random variant*

1. Disable the variant override 
1. Do a new install of the app
1. Confirm variant added to atb requests
1. Repeat several times to confirm that variant is random


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
